doc/Doxyfile

```diff
@@ -1013,7 +1013,7 @@ EXAMPLE_PATTERNS       =
 # irrespective of the value of the RECURSIVE tag.
 # The default value is: NO.
 
-EXAMPLE_RECURSIVE      = NO
```

Comment:
The first one is for making it recursively search for examples when you provide it with an example directory and try to include some example code using `\include`. I never got it working or integrated with this PR, but we definitely don't want to have to include every example directory in this Doxygen config file when we do later.

The last one is to stop enums previews from blowing up the Doxygen when you have a massive enum declaration... They look horrible, and it's much cleaner and better looking to just have the enum name, brief description, then click on it to see the individual values rather than cramming them all into the index.

Reply:
The final addition tells Doxygen to ignore the `__packed` macro, because it tries to add it to the function declaration in the docs without resolving it, messing things up and looking wrong.